tests/utilities/test_collections.py

```diff
@@ -328,6 +328,19 @@ class RandomPydantic(pydantic.BaseModel):
             output_model.val == input_model.val
         ), "The fields value should be used, not the default factory"
 
+    def test_visit_collection_remembers_unset_pydantic_fields(self):
+        class RandomPydantic(pydantic.BaseModel):
+            val: uuid.UUID = pydantic.Field(default_factory=uuid.uuid4)
```

Comment:
I like this test a lot, but I kinda want to see this `RandomPydantic` class have more than one field, and then make assertions on a mix of set and unset fields.  I'm pretty sure this test is correct, but I think it could demonstrate it more clearly.

Reply:
Good point -- I went ahead and added one required field and one optional field (of a different type) and am setting a plain Python default (instead of using `pydantic.Field`) to make sure our bases are covered. Let me know if there's an extra case that might be interesting.